message Car {
  required float position_m = 1;
  required float velocity_m_s = 2;
}

message Client {
  required string host = 1;
  required uint32 port = 2;
}

message Config {
  required float length_m = 1;
  required float start_spacing_m = 2;
  repeated Client clients = 3;
  required float time_step_s = 4;
  required float simulation_duration_s = 5;
  required float max_velocity_m_s = 6;
}

message DriveRequest {
  required Car current = 1;
  required Car next = 2;
  required float time_s = 3;
  required float reward = 4;
}

message DriveResponse {
  required float acceleration_m_s2 = 1;
}
